secrets/onepassword/client.go

```diff
@@ -1,78 +1,40 @@
 package onepassword
 
 import (
-	"bytes"
 	"fmt"
-	"os"
-	"os/exec"
-	"regexp"
 
-	"github.com/kr/pty"
+	op "github.com/ameier38/onepassword"
+	"github.com/pkg/errors"
 )
 
-type OnePasswordClient interface {
-	Get(key string) string
+type Client interface {
 	SignIn(domain string, email string, secretKey string, masterPassword string) error
+	Get(vault string, key string) (string, error)
 }
 
-type OnePasswordCliClient struct {
+type OP struct {
+	OP *op.Client
 }
 
-func (c OnePasswordCliClient) SignIn(domain string, email string, secretKey string, masterPassword string) error {
-	fmt.Println("Signing into 1password.")
-
-	cmd := exec.Command("/usr/local/bin/op", "signin", domain, email)
-	var outb bytes.Buffer
-	cmd.Stdout = &outb
-	cmd.Stderr = os.Stderr
-
-	b, err := pty.Start(cmd)
+func (client *OP) SignIn(domain string, email string, secretKey string, masterPassword string) error {
+	op, err := op.NewClient("/usr/local/bin/op", domain, email, masterPassword, secretKey)
 	if err != nil {
-		fmt.Println(err, "/usr/local/bin/op signin failed with %s")
-		return err
-	}
-
-	go func() {
-		b.Write([]byte(secretKey + "\n"))
-		b.Write([]byte{4})
-		b.Write([]byte{4})
-		b.Write([]byte(masterPassword + "\n"))
-		b.Write([]byte{4})
-		b.Write([]byte{4})
-	}()
-
-	fmt.Println("Started '/usr/local/bin/op'.")
-
-	cmd.Wait()
-
-	r, _ := regexp.Compile("export OP_SESSION_(.+)=\"(.+)\"")
-	matches := r.FindAllStringSubmatch(outb.String(), -1)
-
-	if len(matches) == 0 {
-		fmt.Println("Could not retrieve token from 1password.")
-		return nil
+		return errors.Wrap(err, "op signin failed")
 	}
 
-	session := matches[0][1]
-	token := matches[0][2]
-	fmt.Println("\nUpdated 'OP_SESSION_externalsecretoperator' environment variable.")
-	os.Setenv("OP_SESSION_"+session, token)
+	client.OP = op
 
 	return nil
 }
 
-// Invoke $ op get item 'key'
-func (c OnePasswordCliClient) Get(key string) string {
-	cmd := exec.Command("/usr/local/bin/op", "get", "item", key)
-	var stdout, stderr bytes.Buffer
-	cmd.Stdout = &stdout
-	cmd.Stderr = &stderr
-	err := cmd.Run()
+func (client *OP) Get(vault string, key string) (string, error) {
+	itemMap, err := client.OP.GetItem(op.VaultName(vault), op.ItemName(key))
+	if itemMap != nil {
+		return "", fmt.Errorf("could not retrieve 1password item '" + key + "'.")
```

Comment:
use `%s` please

Reply:
Fixed in https://github.com/ContainerSolutions/externalsecret-operator/commit/a90f3d02e349a6ce8a9de027b35aa653f60cf369